nca/Utils/ExplTracker.py

```diff
@@ -0,0 +1,419 @@
+        if self.ExplDescriptorContainer.get(peer_name):
+            if not self.ExplPeerToPolicyContainer.get(peer_name):
+                self.ExplPeerToPolicyContainer[peer_name] = self.ExplPolicies()
+            self.ExplPeerToPolicyContainer[peer_name].add_policy(policy_name,
+                                                                 egress_dst,
+                                                                 ingress_src,
+                                                                 )
+
+    @staticmethod
+    def extract_peers(conns):
+        """
+        Utility function to extract the peer names held in a connectivity element
+        :param ConnectivityProperties conns:
+        :return: PeerSet src_peers, PeerSet dst_peers: sets of collected peers
+        """
+        src_peers = PeerSet()
+        dst_peers = PeerSet()
+        for cube in conns:
+            conn_cube = conns.get_connectivity_cube(cube)
+            src_peers |= conn_cube["src_peers"]
+            dst_peers |= conn_cube["dst_peers"]
+        return src_peers, dst_peers
+
+    def set_connections_and_peers(self, conns, peers):
+        """
+        Update the calculated connections and topology peers into ExplTracker
+        :param ConnectivityProperties conns: the connectivity mapping calculated by the query
+        :param PeerSet peers: all the peers in the container
+        """
+        self.all_conns = conns
+        self.all_peers = peers
+        # add all missing 'special' peers (like 0.0.0.0/0) with default policy.
+        for peer in self.all_peers:
+            peer_name = peer.full_name()
+            if not self.ExplPeerToPolicyContainer.get(peer_name):
+                if not self.ExplDescriptorContainer.get(peer_name):
+                    self.add_item('', 0, peer_name)
+                self.add_default_policy(PeerSet([peer]), peers, False)
+                self.add_default_policy(peers, PeerSet([peer]), True)
+
+    def are_peers_connected(self, src, dst):
+        """
+        Check if a given pair of peers are connected
+        :param str src: name of the source peer
+        :param str dst: name of the destination peer
+        :return: bool: True for connected, False for disconnected
+        """
+        if not self.all_conns:
+            NcaLogger().log_message('Explainability error: Connections were not set yet, but peer query was called', level='E')
+        for cube in self.all_conns:
+            src_peers_names = []
+            dst_peers_names = []
+            conn_cube = self.all_conns.get_connectivity_cube(cube)
+            src_peers = conn_cube['src_peers']
+            dst_peers = conn_cube['dst_peers']
+            for peer in src_peers:
+                src_peers_names.append(peer.full_name())
+            for peer in dst_peers:
+                dst_peers_names.append(peer.full_name())
+            if src in src_peers_names and dst in dst_peers_names:
+                return True
+        return False
```

Comment:
`return True if self.all_conns & ConnectivityProperties.make_conn_props_from_dict({"src_peers": PeerSet({src}), "dst_peers": PeerSet({dst})) else False`
```suggestion
        for cube in self.all_conns:
            src_peers_names = []
            dst_peers_names = []
            conn_cube = self.all_conns.get_connectivity_cube(cube)
            src_peers = conn_cube['src_peers']
            dst_peers = conn_cube['dst_peers']
            for peer in src_peers:
                src_peers_names.append(peer.full_name())
            for peer in dst_peers:
                dst_peers_names.append(peer.full_name())
            if src in src_peers_names and dst in dst_peers_names:
                return True
        return False
```